heat/core/manipulations.py

```diff
@@ -781,6 +783,350 @@ def hstack(tup):
+            # CASE 2: padding in split dimension and function runs on more than 1 process
+            #
+            # Pad only first/last tensor portion on node (i.e. only beginning/end in split dimension)
+            # --> "Calculate" pad tuple for the corresponding tensor portion/ the two indices which have to be set to zero
+            #      in different paddings depending on the dimension
+            #       Calculate the index of the first element in tuple that has to change/set to zero in
+            #       some dimensions (the following is the second)
+            # ------------------------------------------------------------------------------------------------------------------
+
+            pad_beginning_list = list(pad)
+            pad_end_list = list(pad)
+            pad_middle_list = list(pad)
+
+            # calculate the corresponding pad tuples
+
+            first_idx_set_zero = 2 * (rank_array - array.split - 1)
+
+            pad_end_list[first_idx_set_zero] = 0
+            pad_beginning_list[first_idx_set_zero + 1] = 0
+            pad_middle_list[first_idx_set_zero : first_idx_set_zero + 2] = [0, 0]
+
+            pad_beginning = tuple(pad_beginning_list)
+            pad_end = tuple(pad_end_list)
+            pad_middle = tuple(pad_middle_list)
+
+            if amount_of_processes >= array.shape[array.split]:
+                last_ps_with_data = array.shape[array.split] - 1
+            else:
+                last_ps_with_data = amount_of_processes - 1
+
+            rank = array.comm.rank
+
+            # first process - pad beginning
+            if rank == 0:
+                pad_tuple_curr_rank = pad_beginning
+
+            # last process - pad end
+            elif rank == last_ps_with_data:
+                pad_tuple_curr_rank = pad_end
+
+            # pad middle
+            else:
+                pad_tuple_curr_rank = pad_middle
+
+            if isinstance(values, int) or isinstance(values, float):
+                padded_torch_tensor = torch.nn.functional.pad(
+                    array_torch, pad_tuple_curr_rank, mode, values
+                )
+
+            elif len(values) == 1 and (isinstance(values[0], int) or isinstance(values[0], float)):
+                padded_torch_tensor = torch.nn.functional.pad(
+                    array_torch, pad_tuple_curr_rank, mode, values[0]
+                )
+
+            else:
+                padded_torch_tensor = array_torch
+                for i in range(len(value_tuple) - 1, -1, -1):
+                    pad_list = [0] * 2 * rank_array
+                    pad_list[i] = pad_tuple_curr_rank[i]
+                    pad_tuple = tuple(pad_list)
+                    padded_torch_tensor = torch.nn.functional.pad(
+                        padded_torch_tensor, pad_tuple, mode, value_tuple[i]
+                    )
+
+    padded_tensor = dndarray.DNDarray(
```

Comment:
can you use `factories.array` here with the `is_split` parameter instead? It should take care of any errant type issues

Reply:
@coquelin77 As far as I recall @ClaudiaComito advised me against it as it might cause problems in case of local empty tensors

Reply:
Indeed I had problems with `factories.array` in the past when a DNDarray accidentally contained no data on some rank. In the meantime I've found that the problem was with the lshape of the DNDarray containing no data not being defined consistently with the rest of the array. Since you define a consistent lshape at line 1157, `factories.array` with the `is_split`attribute is probably the way to go.